### PR TITLE
Adds required jobs support to gamemodes

### LINKED ISSUE #18106: Revolution rounds can happen without heads of staff in round start.
So, a round in Sybil started, it was revolution, but then this happened:

> ADMIN LOG: The roundtype (revolution) has no antagonists, continuous round has been defaulted to on and midround_antag has been defaulted to off.

viva

No heads but still a rev.

### PR DESCRIPTION
**Pull request by AnturK**
Fixes #18106

I don't really like this but alternatives are big refactors.

### CODE REVIEW

**Comment by Fikou** [comment]
maybe cult rounds should require a chaplain or any kind of implanted personnel since they are the sacrifices unless it has something like that already

**Comment by Eaglendia** [comment]
It doesn't. I'm pretty sure it just picks a random person in that case. I've seen min-pop cults have to sac random assistants and miners before.

File: code/game/gamemodes/revolution/revolution.dm [hunk 1]
```diff
@@ -14,6 +14,7 @@
 	antag_flag = ROLE_REV
 	false_report_weight = 10
 	restricted_jobs = list("Security Officer", "Warden", "Detective", "AI", "Cyborg","Captain", "Head of Personnel", "Head of Security", "Chief Engineer", "Research Director", "Chief Medical Officer")
+	required_jobs = list(list("Captain"=1),list("Head of Personnel"=1),list("Head of Security"=1),list("Chief Engineer"=1),list("Research Director"=1),list("Chief Medical Officer"=1)) //Any head present
```
Comment (Cyberboss) [inline code/game/gamemodes/revolution/revolution.dm]: These strings aren't `#define`d anywhere?

File: code/controllers/subsystem/job.dm [hunk 2]
```diff
@@ -354,7 +354,27 @@ SUBSYSTEM_DEF(job)
...
+					return FALSE
+				if(J.current_positions < required_group[rank])
+					group_ok = FALSE
+					break
+			if(group_ok)
+				return TRUE
+		SSticker.mode.setup_error = "Required jobs not present."
+		return FALSE
+	else
+		return TRUE
```
Comment (Cyberboss) [inline code/controllers/subsystem/job.dm]: Early return on `if(required_jobs.len)`

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/gamemodes/revolution/revolution.dm (hunk 1 above)
Comment (AnturK) [reply code/game/gamemodes/revolution/revolution.dm]: Nope, that's why i added typo check. Honestly most job checks should use define id's or straight checking types but consistency of these is a mess.

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**